src/app/rundown-view/components/rundown-tooltips/piece-tooltip/piece-tooltip.component.ts

```diff
@@ -46,10 +56,30 @@ export class PieceTooltipComponent {
     return piece.metadata.type
   }
 
-  constructor(private readonly changeDetectorRef: ChangeDetectorRef) {}
+  constructor(
+    private readonly changeDetectorRef: ChangeDetectorRef,
+    private readonly tv2PieceTooltipContentFieldService: Tv2PieceTooltipContentFieldService
+  ) {}
+
+  public ngOnChanges(changes: SimpleChanges): void {
+    const pieceChange: SimpleChange | undefined = changes['piece']
+    const mediaChange: SimpleChange | undefined = changes['media']
+    if (pieceChange || mediaChange) {
+      this.updateTooltipContent()
+    }
+  }
+
+  public updateTooltipContent(): void {
+    this.tooltipContentFields = this.tv2PieceTooltipContentFieldService.getTooltipContentForPiece(this.piece, this.media, this.durationInMs)
+  }
 
   public get shouldShowHoverScrub(): boolean {
     const tv2Piece: Tv2Piece = this.piece as Tv2Piece
-    return tv2Piece.metadata?.type === Tv2PieceType.VIDEO_CLIP || tv2Piece.metadata?.type === Tv2PieceType.JINGLE
+    return (tv2Piece.metadata?.type === Tv2PieceType.VIDEO_CLIP || tv2Piece.metadata?.type === Tv2PieceType.JINGLE) && tv2Piece.metadata.sourceName !== undefined
+  }
```

Comment:
Since this only changes when the piece does, it could be changed into a cached value, like the tooltip content fields.

Reply:
It is now a cached value!